hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -129,6 +137,7 @@ public void snapshotState() {
     // it would check the validity.
     // wait for the buffer data flush out and request a new instant
     flushRemaining(false);
+    this.updateStrategy.reset();
   }
```

Comment:
Can we move all the consistent hashing bucket logic down to the sub-clazz ?

Reply:
This is not only for consistent hashing bucket index. It also handles concurrent clustering for a normal stream write, i.e., reject updates to file groups that are under clustering.